File: my_automation/automation/log_in_testing.py
import unittest
from selenium import webdriver
from selenium.webdriver import Keys
from selenium.webdriver.chrome.options import Options
from selenium.webdriver.common.by import By




class My_Log_In_Test(unittest.TestCase):

    @classmethod
    def setUpClass(cls):
        options = Options()
        options.add_experimental_option("detach", True)
        options.add_argument("--disable-blink-features=AutomationControlled")
        options.add_argument("enable-automation")
        options.add_argument("--start-maximized")
        options.add_argument("--lang=en-GB")
        cls.driver = webdriver.Chrome(options)
        cls.driver.implicitly_wait(10)
        cls.driver.get("https://practicetestautomation.com/practice-test-login/")




    def test_1_positive_logging(self):
        driver = self.driver
        userName = driver.find_element(by=By.XPATH, value="//input[@id='username']")
        userName.send_keys("student")
        userName.send_keys(Keys.ENTER)

        password = driver.find_element(by=By.ID, value="password")
        password.send_keys("Password123")
        password.send_keys(Keys.ENTER)

        driver.find_element(by=By.ID, value="submit").click()



        print("========== TEST 1 POSITIVE LOG IN TEST ==========")
        actualText = "Logged In Successfully"
        actualURL = "https://practicetestautomation.com/logged-in-successfully/"
        expectedText = driver.find_element(by=By.XPATH, value="//h1[@class='post-title']")
        value = expectedText.text

        print("ERROR MESSAGE: ----> " + str(expectedText.is_displayed()))
        print("URL VALUE:     ----> "+ str(driver.current_url))

        driver.find_element(by=By.XPATH, value="//a[@class='wp-block-button__link has-text-color has-background has-very-dark-gray-background-color']").click()
        expectedCurrentURL = driver.current_url
        if expectedCurrentURL == actualURL:
            print("EXPECTED AND ACTUAL URLS MATCHING: ---> |=>" + expectedCurrentURL + "<=| <--- This one its value!")
        else:
            print("EXPECTED AND ACTUAL URLS MATCHING: ---> |=>" + expectedCurrentURL + "<=| <--- This one its value!")

        if expectedText == actualText:
            print("EXPECTED TEXT AND ACTUAL IS MATCHING |=>" + value + "<=| <---- This one its value")
        else:
            print("EXPECTED TEXT AND ACTUAL IS NOT MATCHING |=>" + value + "<=| <---- This one its value")


    def test_2_negative_username(self):
        driver = self.driver
        incorrectUserName = driver.find_element(by=By.XPATH, value="//input[@id='username']")
        incorrectUserName.send_keys("incorrectUser")
        incorrectUserName.send_keys(Keys.ENTER)

        negativePassword = driver.find_element(by=By.ID, value="password")
        negativePassword.send_keys("Password123")
        negativePassword.send_keys(Keys.ENTER)

        driver.find_element(by=By.ID, value="submit").click()

        print("\n========== TEST 2 NEGATIVE USERNAME TEST ==========")
        actualUserNameText = "Your username is invalid!"
        expectedErrorMessage = driver.find_element(By.XPATH, "//*[@class='show']")
        print("ERROR MESSAGE: ----> " + str(expectedErrorMessage.is_displayed()))
        print("TEXT VALUE:     ----> " + str(expectedErrorMessage.text))
        driver.find_element(By.ID, "submit").click()
        if expectedErrorMessage == actualUserNameText:
            print("EXPECTED TEXT AND ACTUAL IS MATCHING |=>" + str(expectedErrorMessage.text) + "<=| <---- This one its value!")
        else:
            print("EXPECTED TEXT AND ACTUAL IS NOT MATCHING |=>" + str(expectedErrorMessage.text) + "<=| <---- This one its value!")

    def test_3_negative_password(self):
        driver = self.driver
        correctUserName = driver.find_element(by=By.XPATH, value="//input[@id='username']")
        correctUserName.send_keys("student")
        correctUserName.send_keys(Keys.ENTER)

        incorrectPassword = driver.find_element(by=By.ID, value="password")
        incorrectPassword.send_keys("incorrectPassword")
        incorrectPassword.send_keys(Keys.ENTER)

        driver.find_element(by=By.ID, value="submit").click()

        print("\n========== TEST 3 NEGATIVE PASSWORD TEST ==========")
        actualInavlidPasswordText = "Your password is invalid!"
        expectedIncorrectPassword = driver.find_element(By.XPATH, "//div[@class='show']")
        print("ERROR MESSAGE: ----> " + str(expectedIncorrectPassword.is_displayed()))
        print("TEXT VALUE:     ----> " + str(expectedIncorrectPassword.text))

        if expectedIncorrectPassword == actualInavlidPasswordText:
            print("EXPECTED TEXT AND ACTUAL IS MATCHING |=>" + str(expectedIncorrectPassword.text) + "<=| <---- This one its value!")
        else:
            print("EXPECTED TEXT AND ACTUAL IS NOT MATCHING |=>" + str(expectedIncorrectPassword.text) + "<=| <---- This one its value!")



    @classmethod
    def tearDownClass(cls):
        """Runs once after all test cases"""
        if cls.driver:
            cls.driver.quit()
            print("Browser closed successfully.")


if __name__ == '__main__':
    unittest.main()

File: log_in_testing.cs
using NUnit.Framework;
using System.Reflection.PortableExecutable;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Interactions;
using static System.Net.Mime.MediaTypeNames;

namespace TestProject1;

public class log_in_testing
{

    IWebDriver driver;
    ChromeOptions options;
    [SetUp]
    public void Setup()
    {
        options = new ChromeOptions();
        options.AddArgument("--disable-blink-features=AutomationControlled");
        options.AddExcludedArgument("enable-automation");
        options.AddArgument("--start-maximized");
        options.AddArguments("--lang=en-GB");
        driver = new ChromeDriver(options);
        driver.Navigate().GoToUrl("https://practicetestautomation.com/practice-test-login/");
    }

    [Test, Order(1)]
    public void Positive_Log_In_Testing()
    {
        
        IWebElement userName = driver.FindElement(By.XPath("//input[@id='username']"));
        IWebElement positivePassword = driver.FindElement(By.XPath("//*[@id='password']"));
        userName.SendKeys("student");
        positivePassword.SendKeys("Password123");
        driver.FindElement(By.XPath("//*[@id='submit']")).Click();
        IWebElement texting = driver.FindElement(By.XPath("//h1[@class='post-title']"));
        string actualText = "Logged In Successfully";
        string expectedText = driver.FindElement(By.XPath("//*[@class='post-title']")).Text;
        string actualURL = "https://practicetestautomation.com/logged-in-successfully/";
        string expectedURL = driver.Url;
        Console.WriteLine("ERROR MESSAGE: " + texting.Displayed);
        Console.WriteLine("EXPECTED TEXT VALUE IS: ----> |" + expectedText + "| <---- This is TEXT value!");
        Console.WriteLine("URL's value: ----> |"+expectedURL+"| <--- This is URL value!");
        driver.FindElement(By.XPath("//a[@class='wp-block-button__link has-text-color has-background has-very-dark-gray-background-color']")).Click();

        if (expectedText.Equals(actualText)) 
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS MATCHING TO EACH OTHER: ---> |" + expectedText + "| <---- This is value of EXPECTED TEXT!");
        }
        else
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS NOT MATCHING TO EACH OTHER: ---> | " + expectedText + " | < ----This is value of EXPECTED TEXT!");
        }

        if (expectedURL.Equals(actualURL))
        {
            Console.WriteLine("EXPECTED AND ACTUAL URL IS MATCHING TO EACH OTHER: ---> |" + expectedURL + "| <---- This is value of EXPECTED URL!");
        }
        else
        {
            Console.WriteLine("EXPECTED AND ACTUAL URL IS NOT MATCHING TO EACH OTHER: ---> |" + expectedURL + "| <---- This is value of EXPECTED URL!");
        }
    }

    [Test, Order(2)]
    public void Negative_UserName_Test()
    {
        IWebElement incorrectUserName = driver.FindElement(By.XPath("//input[@id='username']"));
        IWebElement positivePassword = driver.FindElement(By.XPath("//*[@id='password']"));
        incorrectUserName.SendKeys("incorrectUser ");
        positivePassword.SendKeys("Password123");
        driver.FindElement(By.XPath("//*[@id='submit']")).Click();
        IWebElement invalidUserName = driver.FindElement(By.XPath("//div[@id='error']"));
        string actualInvalidUserName = "Your username is invalid!";
        var expectedInvalidUserName = driver.FindElement(By.XPath("//div[@id='error']"));
        string messageOfExpectedInavlidUserName = expectedInvalidUserName.Text;
        Console.WriteLine("ERROR MESSAGE: ---> " + invalidUserName.Displayed);
        Console.WriteLine("EXPECTATION OF MESSAGE: ----> |" + messageOfExpectedInavlidUserName + "| <---- This is the value of EXPECTED WARNING!!!");

        if (messageOfExpectedInavlidUserName.Equals(actualInvalidUserName))
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS MATCHING TO EACH OTHER: ----> |" + messageOfExpectedInavlidUserName + "| <---- This is the value of EXPECTED TEXT!!!");
        }
        else
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS MATCHING NOT TO EACH OTHER: ----> |" + messageOfExpectedInavlidUserName + "| <---- This is the value of EXPECTED TEXT!!!");
        }

    }

    [Test, Order(3)]
    public void Negative_Password_Test()
    {
        IWebElement correctUserName = driver.FindElement(By.XPath("//input[@id='username']"));
        IWebElement incorrectPassword = driver.FindElement(By.XPath("//*[@id='password']"));
        correctUserName.SendKeys("student");
        incorrectPassword.SendKeys("incorrectPassword");
        driver.FindElement(By.XPath("//*[@id='submit']")).Click();
        IWebElement invalidUserName = driver.FindElement(By.XPath("//div[@id='error']"));
        string actualInvalidPassword = "Your password is invalid!";
        var expectedInvalidPassword = driver.FindElement(By.XPath("//div[@class='show']"));
        string messageOfExpectedInavlidPassword = expectedInvalidPassword.Text;
        Console.WriteLine("ERROR MESSAGE: ---> " + invalidUserName.Displayed);
        Console.WriteLine("EXPECTATION OF MESSAGE: ----> |" + messageOfExpectedInavlidPassword + "| <---- This is the value of EXPECTED WARNING!!!");

        if (messageOfExpectedInavlidPassword.Equals(actualInvalidPassword))
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS MATCHING TO EACH OTHER: ----> |" + messageOfExpectedInavlidPassword + "| <---- This is the value of EXPECTED TEXT!!!");
        }
        else
        {
            Console.WriteLine("EXPECTED AND ACTUAL TEXT IS MATCHING NOT TO EACH OTHER: ----> |" + messageOfExpectedInavlidPassword + "| <---- This is the value of EXPECTED TEXT!!!");
        }
    }

    [TearDown]
    public void tearDown()
    {
        driver.Quit();
    }
}
